File: 2022/16/javascript/main.mjs
import createLayout from "./layout.mjs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";

// --- Read and parse the input file ---
let layout = createLayout(INPUT_FILE);

// --- Puzzle 1 ---
let total = layout.dfs1();
console.log("1. Total pressure released:", total);

// --- Puzzle 2 ---
total = layout.dfs2();
console.log("2. Total pressure released with an elephant:", total);

File: 2022/16/python/main.py
from lib import Layout

# --- Constants ---
INPUT_FILE = "../input/input.txt"

layout = Layout(INPUT_FILE)

# ---  Puzzle 1 ---
total = layout.dfs1()
print(f"1. Total pressuere released: {total:,d}")

# --- Puzzle 2 ---
total = total = layout.dfs2()
print(f"2. Total pressure released with an elephant: {total:,d}")
